src/common/JSONImporter.js

```diff
@@ -28,23 +69,10 @@ define([
         }
 
         async toJSON(node, omit=new Set()) {
-            if(!omit) {
-                omit = new Set();
-            } // Sentinel
-
-            if(typeof omit === 'string') {
-                omit = new Set([omit]);
-            }
-
-            if(typeof omit === 'boolean' && omit === true) {
-                omit = new Set(['children']);
-            } // Backwards compatible with shallow
-
-            if(Array.isArray(omit)) {
-                omit = new Set(omit);
-            }
-
-            ['id', 'path', 'guid'].forEach(attr => omit.delete(attr));
+            const attributesFilter = new NodeAttributesFilter(omit, {
```

Comment:
I would opt for making the class public so the user passes `omit` with the correct type rather than passing a generic set. Also, it would be nice not to be constructing this on each call to `toJSON` as it is recursive. For backwards compatibility, you could make two methods: `toJSON` and `_toJSON`. The latter could be called internally for the recursive calls and require `omit` to be the custom class. The former could support `omit` being a boolean and just convert it to the correct type and call the internal method.

Reply:
Refactored into separate methods.